sdk/storage/storage-queue/RecordAndPlayback.md

```diff
@@ -0,0 +1,192 @@
+
+## Recordings
+
+- Recordings are being saved in one folder for each describe-block test suite
+  recording file structure
+  - `recordings/node/<describe-block-title>/recording_<test-title>.js` for node tests and
+  - `recordings/browsers/<describe-block-title>/recording_<test-title>.json` for browser tests.
+- The file name of the recording for a test preserves the title of the `describe`-block and the corresponding `it`-block (with special characters appropriated). The recordings for browser tests go into the `./recordings/browsers/` folder(Similarly, node recordings in the `./recordings/node/` folder).
+
+  For example
+
+  ```typescript
+  describe("Aborter", () => {
+    it("should abort when abort() is called", async () => {
+      ...
+      ...
+    });
+  });
+  ```
+
+  (Browser Test) Recording corresponding to the above test is placed at
+  `./recordings/browsers/aborter/recording_should_abort_when_abort_is_called.json`
+
+  [ Following this rule - `./recordings/browsers/<describe-block-title>/recording_<test-title>.json` ]
+
+- Just like the test recordings, we save the requests and responses from the before and after sections of the describe block in - `recordings/{node|browsers}/<describe-block-title>/recording_before_all_hook.{js|json}`.
+
+---
+
+## Updating an existing test/test-suite
+
+- Recordings are saved in `./recordings/` folder.
+- If a test is modified, we might need to record it again in order to equip the recording to accommodate any updates in the function calls that invoke http requests.
+
+  - execute the test/test-suite script(or all the tests) by setting the env variable `TEST_MODE = record`
+
+- If only the test title is updated,
+
+  - we can either execute that specific test again to generate the recording
+
+    or
+
+  - we can just update the file name of the recording accordingly
+
+- **Commit** the generated recordings.
+
+---
+
+## Adding a new test/test-suite
+
+- Any function call that is not in the `it`-block and belongs to a `describe`-block must go in one of the `beforeEach`, `afterEach`, `before` or `after` sections.
+
+- Recordings corresponding to `before` or `after` sections are saved under `recordings/{node|browsers}/<describe-block-title>/recording_before_all_hook.{js|json}`.
+
+- Recordings corresponding to `beforeEach` or `afterEach` sections are saved along with the test recordings(`recordings/{node|browsers}/<describe-block-title>/recording_<test-title>.{js|json}`).
+
+- Follow the below template for adding any new test. `before` and `after` sections are optional, `beforeEach` and `afterEach` sections are compulsory.
+
+  ```typescript
+  import { record } from "./utils/recorder";
+
+  describe("<describe-block-title>", () => {
+    // before section is optional
+    before(async function() {
+      recorder = record(this);
+      /*Place your code here*/
+      recorder.stop();
+    });
+
+    // after section is optional
+    after(async function() {
+      recorder = record(this);
```

Comment:
Recorder is a global variable initialized in `before` section, why do we need to create a recorder here?

Reply:
```typescript
recorder = record(this);
/*Place your code here*/
recorder.stop();
```
- The plan is to record the requests in the `/*Place your code here*/`. 
- We do that by creating the `recorder` and by doing `recorder.stop()`, we save the recording in `recording_before_all_hook.{js|json}` in the record mode.
- Same is the case with `after` section.
- In conclusion, `before` and `after` sections are recorded separately.

`beforeEach` and `afterEach` are included in the test recordings. i.e., we create the recorder in `beforeEach` and stop the recorder in `afterEach`.

Reply:
Added this as part of the document!